Modules/Loadable/Markups/qSlicerMarkupsModuleWidget.cxx

```diff
@@ -1076,12 +1076,9 @@ void qSlicerMarkupsModuleWidget::updateWidgetFromMRML()
   int numberOfPoints = d->MarkupsNode->GetNumberOfControlPoints();
   if (d->activeMarkupTableWidget->rowCount() != numberOfPoints)
     {
-    // clear it out
-    d->activeMarkupTableWidget->setRowCount(numberOfPoints);
-    }
-  for (int m = 0; m < numberOfPoints; m++)
-    {
-    this->updateRow(m);
+    // force full update of the table
+    // (after node change or batch update with multiple rows added or deleted)
+    this->updateRow(-1);
```

Comment:
What about having `updateRow(int controlPointIndex)` and `updateRows()`?

Reply:
OK